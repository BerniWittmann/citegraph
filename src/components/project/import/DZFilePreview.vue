<template>
  <v-sheet class="mb-2" :elevation="2">
    <v-container class="py-1">
      <v-row>
        <v-col cols="auto" class="d-flex justify-center py-0">
          <file-icon class="file-icon"/>
        </v-col>
        <v-col class="py-0">
          <v-list-item three-line>
            <v-list-item-content class="align-self-centerpy-0">
              <v-list-item-title class="title mb-1"><span data-dz-name></span></v-list-item-title>
              <v-list-item-subtitle><span data-dz-size></span></v-list-item-subtitle>
              <v-list-item-subtitle>
                <div class="upload-progress">
                  <div class="upload-progress__background"></div>
                  <div data-dz-uploadprogress class="upload-progress__progress-bar"></div>
                </div>
              </v-list-item-subtitle>
              <v-list-item-subtitle><span class="error--text" data-dz-errormessage></span></v-list-item-subtitle>
            </v-list-item-content>
          </v-list-item>
        </v-col>
        <v-col class="py-0" cols="auto">
          <v-icon @click="() => {}" large data-dz-remove>mdi-close</v-icon>
        </v-col>
      </v-row>
    </v-container>
  </v-sheet>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import FileIcon from '@/assets/svgs/File.vue'

@Component({
  components: {
    FileIcon
  }
})
export default class DZFilePreview extends Vue {
}
</script>

<style lang="scss" scoped>
.file-icon {
  height: 100%;
}

.upload-progress {
  position: relative;
  height: 4px;

  .upload-progress__background, .upload-progress__progress-bar {
    background-color: $accent_color;
    height: inherit;
    border-radius: 4px;
    position: absolute;
  }

  .upload-progress__background {
    width: 100%;
    opacity: 0.15;
  }

  .upload-progress__progress-bar {
    width: 0;
    transition: 0.1s
  }
}

</style>
