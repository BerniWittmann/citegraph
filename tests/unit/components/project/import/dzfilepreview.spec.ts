import { shallowMount } from '@vue/test-utils'
import { i18n } from '../../../setupPlugins'

import DZFilePreview from '@/components/project/import/DZFilePreview.vue'

describe('components/project/import/DZFilePreview.vue', () => {
  function getWrapper () {
    return shallowMount(DZFilePreview, {
      i18n
    })
  }

  it('renders', () => {
    const wrapper = getWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
