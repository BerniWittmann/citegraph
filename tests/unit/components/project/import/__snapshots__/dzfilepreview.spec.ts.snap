// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/project/import/DZFilePreview.vue renders 1`] = `
<v-sheet-stub elevation="2" tag="div" class="mb-2">
  <v-container-stub tag="div" class="py-1">
    <v-row-stub tag="div">
      <v-col-stub cols="auto" tag="div" class="d-flex justify-center py-0">
        <file-icon-stub class="file-icon"></file-icon-stub>
      </v-col-stub>
      <v-col-stub tag="div" class="py-0">
        <v-list-item-stub activeclass="" tag="div" threeline="true">
          <v-list-item-content-stub class="align-self-centerpy-0">
            <v-list-item-title-stub class="title mb-1"><span data-dz-name=""></span></v-list-item-title-stub>
            <v-list-item-subtitle-stub><span data-dz-size=""></span></v-list-item-subtitle-stub>
            <v-list-item-subtitle-stub>
              <div class="upload-progress">
                <div class="upload-progress__background"></div>
                <div data-dz-uploadprogress="" class="upload-progress__progress-bar"></div>
              </div>
            </v-list-item-subtitle-stub>
            <v-list-item-subtitle-stub><span data-dz-errormessage="" class="error--text"></span></v-list-item-subtitle-stub>
          </v-list-item-content-stub>
        </v-list-item-stub>
      </v-col-stub>
      <v-col-stub cols="auto" tag="div" class="py-0">
        <v-icon-stub large="" datadzremove="" data-dz-remove="">mdi-close</v-icon-stub>
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</v-sheet-stub>
`;
